diff --git a/MakeDefs/MakeDefs.cpp b/MakeDefs/MakeDefs.cpp
index 5eb9640..77bfd77 100644
--- a/MakeDefs/MakeDefs.cpp
+++ b/MakeDefs/MakeDefs.cpp
@@ -105,8 +105,11 @@ int main(int argc, char* argv[])
 			name.clear();
 			s = *i;
 
+			const auto comma = s.find(',');
+			if (comma == string::npos) continue;
+
 			bool u = true;
-			for (k = s.find_first_not_of(" \t"); s[k] != ','; k++)
+			for (k = s.find_first_not_of(" \t"); k < comma; k++)
 			{
 				if (s[k] == '_')
 				{
